app/src/main/java/org/mozilla/focus/settings/SettingsFragment.java

```diff
@@ -90,6 +94,40 @@ public boolean onPreferenceTreeClick(PreferenceScreen preferenceScreen, Preferen
             TelemetryWrapper.openSearchSettingsEvent();
         } else if (preference.getKey().equals(resources.getString(R.string.pref_key_screen_autocomplete))) {
             navigateToFragment(new AutocompleteSettingsFragment());
+        } else if (preference.getKey().equals(resources.getString(R.string.pref_key_reset_default_browser))) {
+            // Reset browser settings to default
+            PreferenceManager.getDefaultSharedPreferences(getActivity().getApplicationContext())
+                    .edit()
+                    .clear()
+                    .apply();
```

Comment:
Just thinking, because this line is writing (clearing) the shared preferences, then it should be synchronized. Otherwise race conditions could occurs.